src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -475,20 +475,13 @@ else if (variableDeclarationFrameType == FrameType.CTOR_FRAME
                  && !isUserDefinedArrangementOfThis(variableDeclarationFrame, ast)) {
             frameWhereViolationIsFound = findFrame(ast, true);
         }
-        else if (variableDeclarationFrameType == FrameType.BLOCK_FRAME) {
-            if (isOverlappingByLocalVariable(ast)) {
-                if (canAssignValueToClassField(ast)
-                        && !isUserDefinedArrangementOfThis(variableDeclarationFrame, ast)
-                        && !isReturnedVariable(variableDeclarationFrame, ast)
-                        && canBeReferencedFromStaticContext(ast)) {
-                    frameWhereViolationIsFound = findFrame(ast, true);
-                }
-            }
-            else if (!validateOnlyOverlapping
-                     && prevSibling == null
-                     && isAssignToken(ast.getParent().getType())
-                     && !isUserDefinedArrangementOfThis(variableDeclarationFrame, ast)
-                     && canBeReferencedFromStaticContext(ast)) {
+        else {
```

Comment:
There's no need for braces, use `else if {` construction as it is done above in this method.